xwiki-platform-core/xwiki-platform-template/xwiki-platform-template-api/src/main/java/org/xwiki/template/TemplateRequirement.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.template;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Component in charge of validation a specific requirement for a given template.
+ * 
+ * @version $Id$
+ * @since 15.0RC1
+ * @since 14.10.1
+ */
+@Role
+@Unstable
+public interface TemplateRequirement
+{
+    /**
+     * @param requirement the requirement key to check
+     * @param value the requirement value to check
+     * @param template the template on which the requirement applies
+     */
+    void checkRequirement(String requirement, String value, Template template) throws TemplateRequirementException;
```

Comment:
Missing doc for the exception

Reply:
Not sure to understand why you specified the first parameter: right now it's always the same as the hint if I'm correct no?

Reply:
> Missing doc for the exception

Indeed, I'm surprised Checkstyle did not complain, I though were had checks for that.